### DISCUSSION TITLE
trouble adding a Teleport server to my teleport

### DISCUSSION THREAD

**Discussion by saliou44** [gravitational/teleport#48437] [Q&A] [answered]
![Capture d’écran 2024-10-21 040641](https://github.com/user-attachments/assets/7b826b52-6a4f-4b96-b3c1-1036ca1f9556)

![Capture d’écran 2024-11-05 150939](https://github.com/user-attachments/assets/bf9d7667-d23b-4658-aae9-77bdc952e953)

my configuration file

![Capture d’écran 2024-11-05 150548](https://github.com/user-attachments/assets/ef4f0a43-fcaf-463f-a115-34dae54eb07f)
![Capture d’écran 2024-11-05 150714](https://github.com/user-attachments/assets/d4a8e5bc-48a8-488e-8c86-4767fab5c913)

![Capture d’écran 2024-11-05 151041](https://github.com/user-attachments/assets/ba42b1f3-8b97-451a-b99c-c88f229e442a)

**Comment by webvictim** [comment]
I'm not quite clear what you're trying to do. Can you explain further?

Did you run the script linked in your first screenshot on a host? What happened?

**Reply by saliou44** [reply]
I tried deploying Teleport using Docker Compose. The deployment was successful. However, when I attempt to add an Ubuntu resource, it doesn’t show up in the web interface.
when i run the scrypt:
![Capture d’écran 2024-11-06 100530](https://github.com/user-attachments/assets/ddafe827-5ba5-4513-b156-1bfdc2fd8a76)

**Reply by webvictim** [reply]
That screenshot shows that the agent has connected to the cluster and is working fine. You can see it in `tctl get nodes`.

If it doesn't show up in the web interface, it's likely a permissions issue which is preventing you from seeing the node. What roles do you have associated with your Teleport user, and what is their contents?

**Reply by saliou44** [reply]
My teleport user have a editor role.
kind: role
metadata:
  description: Edit cluster configuration
  labels:
    teleport.internal/resource-type: preset
  name: editor
  revision: 61978ee9-6921-4d84-9ce9-a1d0b16b5063
spec:
  allow:
    rules:
    - resources:
      - user
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - role
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - bot
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - crown_jewel
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - db_object_import_rule
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - oidc
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - saml
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - github
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - oidc_request
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - saml_request
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - github_request
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - cluster_audit_config
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - cluster_auth_preference
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - auth_connector
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - cluster_name
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - cluster_networking_config
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - session_recording_config
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - external_audit_storage
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - ui_config
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - trusted_cluster
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - remote_cluster
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - token
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - connection_diagnostic
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - db
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - database_certificate
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - installer
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - device
      verbs:
      - list
      - create
      - read
      - update
      - delete
      - create_enroll_token
      - enroll
    - resources:
      - db_service
      verbs:
      - list
      - read
    - resources:
      - instance
      verbs:
      - list
      - read
    - resources:
      - login_rule
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - saml_idp_service_provider
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - user_group
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - plugin
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - okta_import_rule
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - okta_assignment
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - lock
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - integration
      verbs:
      - list
      - create
      - read
      - update
      - delete
      - use
    - resources:
      - billing
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - cluster_alert
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - access_list
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - node
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - discovery_config
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - security_report
      verbs:
      - list
      - create
      - read
      - update
      - delete
      - use
    - resources:
      - audit_query
      verbs:
      - list
      - create
      - read
      - update
      - delete
      - use
    - resources:
      - access_graph
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - server_info
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - access_monitoring_rule
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - app_server
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - vnet_config
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - bot_instance
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - access_graph_settings
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - spiffe_federation
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - notification
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - static_host_user
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - user_task
      verbs:
      - list
      - create
      - read
      - update
      - delete
  deny: {}
  options:
    cert_format: standard
    create_db_user: false
    create_desktop_user: false
    desktop_clipboard: true
    desktop_directory_sharing: true
    enhanced_recording:
    - command
    - network
    forward_agent: true
    idp:
      saml:
        enabled: true
    max_session_ttl: 30h0m0s
    pin_source_ip: false
    port_forwarding: true
    record_session:
      desktop: false
    ssh_file_copy: true
version: v7

**Comment by saliou44** [comment]
its okey i create a new role 
admin and i have access

**Reply by saliou44** [reply, marked as answer]
kind: role
version: v4
metadata:
  name: admin
spec:
  options:
    max_session_ttl: 24h
    require_password: true      # Active l'obligation de mot de passe
  allow:
    logins: ['root','noc']   # Utilisateurs Unix autorisés
    node_labels:
      '*': '*'                  # Accès à tous les nœuds
    rules:
      - resources: ['*']
        verbs: ['*']            # Donne accès à toutes les actions sur toutes les ressources

**Reply by webvictim** [reply]
Yes, adding the `logins` field is what did this for you :) Glad it works now!

**Reply by saliou44** [reply]
Thks!
